Client/Frontend/Browser/TabGroupData.swift

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import MozillaAppServices
+
+public enum TabGroupTimerState: String, Codable {
+    case navSearchLoaded
+    case tabNavigatedToDifferentUrl
+    case tabSwitched
+    case tabSelected
+    case newTab
+    case openInNewTab
+    case openURLOnly
+    case none
+}
+
+public class TabGroupData: NSObject, NSCoding {
+    var tabAssociatedSearchTerm: String = ""
+    var tabAssociatedSearchUrl: String = ""
+    var tabAssociatedNextUrl: String = ""
+    var tabHistoryCurrentState = ""
+    // Is not used currently can be removed? Name is confusing with enum
+    var tabGroupTimerState = ""
+
+    func tabHistoryMetadatakey() -> HistoryMetadataKey {
+        return HistoryMetadataKey(url: tabAssociatedSearchUrl, searchTerm: tabAssociatedSearchTerm, referrerUrl: tabAssociatedNextUrl)
+    }
+    
+    var jsonDictionary: [String: Any] {
+            return [
+                "tabAssociatedSearchTerm": String(self.tabAssociatedSearchTerm),
```

Comment:
We could have some keys defined instead of strings I think, since the keys are used in multiple places